src/tblib/pickling_support.py

```diff
@@ -49,8 +62,12 @@ def pickle_exception(obj):
     assert isinstance(rv, tuple)
     assert len(rv) >= 2
 
+    use_builtin = isinstance(obj.__reduce_ex__, BuiltinFunctionType) and \
+                  isinstance(obj.__reduce__, BuiltinFunctionType) and \
+                  not isinstance(obj, IOError)
+
     return (
-        unpickle_exception,
+        unpickle_builtin_exception if use_builtin else unpickler_exception,
```

Comment:
Is this a typo? There is no `unpickler_exception`. Either way, you need to add a bunch of tests cause this should cause failures.

Reply:
Yes, typo. Sorry. Fixed.

Regarding tests - I do not have much time currently, so when time allows, I will add some tests for all situations mentioned in the `use_builtin` check (i.e. custom `__reduce__`, `__reduce_ex__`, `IOError`, “normal” exception).